### PR TITLE
Builder Puppet module to 4.4

### PR DESCRIPTION
**Pull request by c-bordon**
|Related issue|
|---|
|closes https://github.com/wazuh/wazuh-jenkins/issues/4365 |

## Description

The necessary scripts were created for the construction of the Puppet module

## Logs example

Build: https://devel.ci.wazuh.info/view/Packages/job/Packages_builder_cbordon/7/console
## Tests

- Build the package in any supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [ ] Package installation
- [ ] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [ ] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] Package install/remove/install
  - [ ] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

File: puppet_module/build_module.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+help() {
+    echo
+    echo "Usage: $0 [OPTIONS]"
+    echo
+    echo "    -b, --branch <branch>           [Required] wazuh-puppet branch or tag"
+    echo "    -s, --store <path>              [Optional] Set the destination path of package. By default, an output folder will be created."
+    echo "    -h, --help                      Show this help."
+    echo
+    exit $1
+}
```
Comment (rauldpm) [inline puppet_module/build_module.sh:64]: Use the new help format.

File: puppet_module/build_module.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# Wazuh package generator
+# Copyright (C) 2022, Wazuh Inc.
```
Comment (rauldpm) [inline puppet_module/build_module.sh:4]: If new development, it should be 2023

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, see minor changes

File: puppet_module/build_module.sh (hunk 2 above)
Comment (c-bordon) [reply puppet_module/build_module.sh]: Done: 8ff1104df0b5b451cbc8267df4e72a9c60c28fd0

**Review by verdx — APPROVED** [review]
LGTM

**Review by rauldpm — APPROVED** [review]
LGTM

**Review by fcaffieri — APPROVED** [review]
LGTM

**Review by DFolchA — APPROVED** [review]

**Review by teddytpc1 — APPROVED** [review]

**Review by teddytpc1 — APPROVED** [review]

**Pull request merged**